x-pack/platform/plugins/shared/lens/common/transforms/config_builder_stub.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { LensAPIConfig, LensItem } from '../../server/content_management';
+
+export function isNewApiFormat(config: unknown): config is LensAPIConfig {
+  return (config as LensAPIConfig)?.state?.isNewApiFormat;
+}
+
+export const ConfigBuilderStub = {
+  /**
+   * @returns Lens item
+   */
+  in(config: LensAPIConfig): LensItem {
+    return config;
+  },
+
+  /**
+   * @returns Lens API config
+   */
+  out(item: LensItem): LensAPIConfig {
+    return {
+      ...item,
+      state: {
+        ...item.state,
+        isNewApiFormat: true,
```

Comment:
This flag is just temporary to validate conversion until replaced by the full config builder. As is this will only be passed along in the internal API requests not persisted in any way.